gitpod-io/gitpod
# Prebuilds List UI

## Description

Implement the Global Prebuilds List UI.

<img width="1552" alt="image" src="https://github.com/gitpod-io/gitpod/assets/29888641/24c2e4ca-2b32-4ab9-ba71-590852631ca4">

## Related Issue(s)

Fixes EXP-1226

## How to test

1. Visit the preview environment (link in section below)
2. [Join my org](https://ft-global-daef03898b.preview.gitpod-dev.com/orgs/join?inviteId=1cdf6a43-6248-4e5f-8036-912881ee4112) to get some prebuilds
3. In the org dropdown, select `Prebuilds`. You should arrive at the table
4. Play around! Any applied filters should be persisted when refreshing and the <kbd>View</kbd> buttons **should not** (they will be implemented in a later PR, now lead to an invalid route)

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ft-global-daef03898b</li>
	<li><b>🔗 URL</b> - <a href="https://ft-global-daef03898b.preview.gitpod-dev.com/workspaces" target="_blank">ft-global-daef03898b.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - ft-global-prebuild-list-gha.22501</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-ft-global-daef03898b%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
